### PR TITLE
Update extension names in docs

### PR DESCRIPTION
**Pull request by rolfedh**

### CODE REVIEW

File: docs/src/main/asciidoc/rest-client.adoc [hunk 1]
```diff
@@ -296,7 +296,7 @@ public interface Client {
 }
 ----
 
-More information about this annotation can be found on the javadoc of link:https://javadoc.io/doc/io.quarkus/quarkus-rest-client/latest/io/quarkus/rest/client/reactive/ClientFormParam.html[`@ClientFormParam`].
+More information about this annotation can be found on the javadoc of link:https://javadoc.io/doc/io.quarkus/quarkus-resteasy-client/latest/io/quarkus/rest/client/reactive/ClientFormParam.html[`@ClientFormParam`].
```
Comment (rolfedh) [inline docs/src/main/asciidoc/rest-client.adoc:299]: Both of these URLs return Error code 521

**Review by gsmet — CHANGES_REQUESTED** [review]
I don't think this is valid for `main` as `quarkus-rest-client` is a perfectly valid thing.
And the occurrences you replaced should stay as is.

File: docs/src/main/asciidoc/rest-client.adoc (hunk 1 above)
Comment (gsmet) [reply docs/src/main/asciidoc/rest-client.adoc:299]: Yeah javadoc.io appears to be down. I don't know if it's temporary or completely dead.

**Comment by gsmet** [comment]
There might things to adjust in 3.8 though.

Here is the situation:

**3.8**

RESTEasy Classic : `quarkus-resteasy` `quarkus-resteasy-client`
RESTEasy Reactive: `quarkus-resteasy-reactive` `quarkus-rest-client-reactive`

**3.9+**

RESTEasy Classic : `quarkus-resteasy` `quarkus-resteasy-client`
RESTEasy Reactive: `quarkus-rest` `quarkus-rest-client` (and this is our default REST stack)

**Comment by rolfedh** [comment]
> I'm closing this one, I had a discussion with Sheila about the context of these changes.
> 
> Thanks anyway.

You're welcome. Thank you.

**Pull request closed without merging**